astroplan/core.py

```diff
@@ -25,82 +40,137 @@ class Observer(object):
+            that initializes a `~astropy.coordinates.Latitude` object with
+            units in degrees.
 
         elevation : `~astropy.units.Quantity`
             The elevation of the observing location, with respect to sea
             level.
 
-        pressure : `~astropy.units.Quantity`
-            The ambient pressure.
-
         relative_humidity : float
             The ambient relative humidity.
 
         temperature : `~astropy.units.Quantity`
             The ambient temperature.
 
-        timezone : WHAT TYPE IS pytz.timezone ?
-            The local timezone.
+        timezone : str or `datetime.tzinfo`
+            The local timezone, as either an instance of `pytz.timezone()` or
+            the string accepted by `pytz.timezone()`.
 
         description : str
             A short description of the telescope, observatory or observing
             location.
         """
-        raise NotImplementedError()
 
-    def set_environment(self, pressure, relative_humidity, temperature):
-        """
-        Updates the Observer object with environmental conditions.
+        if name is not None:
+            self.name = name
 
-        <longer description>
+        if pressure is None:
+            pressure = 0
 
-        Parameters
-        ----------
-        pressure : `~astropy.units.Quantity`
-            The ambient pressure.
+        if temperature is None:
+            temperature = 0 * u.deg_C
 
-        relative_humidity : `~astropy.units.Quantity`
-            The ambient relative humidity.
+        if relative_humidity is None:
+            relative_humidity = 0
 
-        temperature :
-            The ambient temperature.
-        """
-        raise NotImplementedError()
+        self.pressure = pressure
+        self.temperature = temperature
+        self.relative_humidity = relative_humidity
 
-    def get_date(self, input_date_time, timezone=None):
-        """
-        Builds an object containing date and time information.
+        # If lat/long given instead of EarthLocation, convert them
+        # to EarthLocation
+        if location is None and (latitude is not None and longitude is not None):
+            accepted_latlon_types = [str, Quantity]
+            if (type(latitude) in accepted_latlon_types and
+                type(longitude) in accepted_latlon_types):
+                latitude = Latitude(latitude, unit=u.degree)
+                longitude = Longitude(longitude, unit=u.degree)
+
+            self.location = EarthLocation.from_geodetic(longitude, latitude,
+                                                        elevation)
+
+        elif isinstance(location, EarthLocation):
+            self.location = location
 
-        Default time zone is UTC.
-        If time zone (local or otherwise) requested, then uses `datetime.datetime`
-        and pytz.timezone to convert from UTC.
+        else:
+            raise TypeError('Observatory location must be specified with '
```

Comment:
Make me happy, and put `EarthLocation` as the 1st option here :) -- we want to encourage the OOP!